### ISSUE TITLE
False positive with Text Context (vitest/no-done-callback)

### ISSUE THREAD

**Issue by kevinsandow** [veritem/eslint-plugin-vitest#158]
Text Context gets wrongly detected as done function, see https://vitest.dev/guide/test-context.html)

```
import { test } from 'vitest'

test("access to ctx", (ctx) => {
  // prints name of the test
  console.log(ctx.meta.name);
});
```

Issue fron eslint:
```
warning  Return a promise instead of relying on callback parameter  vitest/no-done-callback
```

I disabled this rule in my project, but I thought I'd let you know...

**Comment by veritem** [comment]
uuuhmm, thanks for let me know about this. I'll push an update

**Comment by charliematters** [comment]
I notice that it handles a value with the name `context`, but still throws warnings when useing `beforeEach` and `afterEach`. We disabled this rule because we used the destructuring method in our tests, and a `ctx` variable in the hooks, all of which fell foul of this rule

**Comment by veritem** [comment]
@charliematters can i have a look at your eslint config?

**Comment by charliematters** [comment]
I've put a repo here which shows the above: https://github.com/charliematters/vitest-eslint


```bash
charlie@charlies-mbp vitest-eslint % npx eslint repro.test.js

/Users/charlie/dev/vitest-eslint/repro.test.js
   4:3   warning  Return a promise instead of relying on callback parameter  vitest/no-done-callback
   8:3   warning  Return a promise instead of relying on callback parameter  vitest/no-done-callback
  12:22  warning  Return a promise instead of relying on callback parameter  vitest/no-done-callback

✖ 3 problems (0 errors, 3 warnings)
```

**Comment by veritem** [comment]
Just found out that `Done Callback` have been deprecated since `v.0.10.0`. This rule have also been causing a lot of issues because it's hard to differentiate done callback from context. I'm gonna go a head and deprecate this rule from now on.

**Comment by charliematters** [comment]
Great - thanks for checking!
